extensions/interactions/CodeRepl/CodeClassifierPredictionService.js

```diff
@@ -0,0 +1,217 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Code classifier prediction functions.
+ *
+ * IMPORTANT NOTE: The prediction function uses the classifier data
+ * of trained model for inference. These functions must be changed if
+ * there are any changes in corresponding classifier training function on
+ * Oppia-ml.
+ */
+
+oppia.factory('CodeClassifierPredictionService', [
```

Comment:
I think this should go into @pranavsid98's PR. Since he is going to manage how prediction services will be detected and loaded. Also classifiers may also require additional services (like code classifier requires SVMPredictionService, CountVectorizerService etc...), how should we keep track of those extra services?

May be maintain a dependency list for each classifier. Also I think it'd be better to not have tight coupling between interaction and classifier. This enables us to use multiple classifiers per interaction. If we name classifier service same as interaction id then we will be able to use only prediction algorithm at a time. Consider a scenario where we upgarde an existing classifier code. In that case, for backward compatibility, if exploration has old version of classifier data then we can still use old prediction API. We can't do this if we name each prediction service with its interaction id. In general, I think, it would be better to name prediction service same with its algorithm id rather than interaction id. What do you think?